### PR TITLE
monitoring: monitor redis and add monitoring user

### PR DESCRIPTION
**Pull request by rerowep**
* Adds monitoring of redis.
* Adds monitoring@rero.ch user for monitoring access.

Co-Authored-by: Peter Weber <peter.weber@rero.ch>

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* rero/rero-ils-ui#<xx>

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Review by iGormilhit — CHANGES_REQUESTED** [review]
Commit message proposition:

```
monitoring: monitor redis and add monitoring user

* Monitors the redis service.
* Adds monitoring@rero.ch user for monitoring access.
* Adds time stamp to monitoring.
```

Changes:
- Rewrites title with verbs (infinitive form).
- Uses the verb monitor instead of add monitoring expression. Redis service instead of redis.
- Adds the **to** in *Adds time stamp to monitoring*.

**Review by iGormilhit — APPROVED** [review]

File: rero_ils/modules/utils.py [hunk 1]
```diff
@@ -419,3 +420,33 @@ def get_patron_from_arguments(**kwargs):
         or Patron.get_patron_by_barcode(kwargs.get('patron_barcode')) \
         or Patron.get_record_by_pid(kwargs.get('patron_pid')) \
         or Patron.get_record_by_pid(kwargs.get('loan').get('patron_pid'))
+
+
+def set_time_stamp(name, **kwargs):
+    """Set time stampe in current cache.
```
Comment (zannkukai) [inline rero_ils/modules/utils.py:426]: typo : I thing that "timestamp" string is a not-splitted word (no need to have "_" between "time" and "stamp"
at L426 --> "stampe" should be "stamp"

File: rero_ils/modules/utils.py [hunk 2]
```diff
@@ -419,3 +420,33 @@ def get_patron_from_arguments(**kwargs):
...
+    time_stamps[name] = {}
+    time_stamps[name]['time'] = utc_now
+    for key, value in kwargs.items():
+        time_stamps[name][key] = value
+    current_cache.set('timestamps', time_stamps)
+    return utc_now
+
+
+def get_time_stamp(name):
+    """Get time stampe in current cache.
```
Comment (zannkukai) [inline rero_ils/modules/utils.py:444]: same here

File: rero_ils/modules/items/tasks.py [hunk 3]
```diff
@@ -64,7 +64,7 @@ def process_late_claimed_issues(
             create_first_claim_count=create_first_claim_count,
             create_next_claim_count=create_next_claim_count
         )
-
+    set_time_stamp('claims-creation', msg=msg)
```
Comment (zannkukai) [inline rero_ils/modules/items/tasks.py:67]: is this function couldn't be a decorator?
something like :
```python
def set_timestamp(func):
    @wraps(func)
    def wrapped(*args, **kwargs):
        result = func(*args, **kwargs)
        # do the job to log the timestamp here with `func.__name__` as parameter 
        return result 
    return wrapped
```

**Review by lauren-d — APPROVED** [review]

**Comment by lauren-d** [comment]
Check test. isort failed !

File: rero_ils/modules/items/tasks.py (hunk 3 above)
Comment (rerowep) [reply rero_ils/modules/items/tasks.py:67]: Will add an additionally function wrapper.

File: rero_ils/config.py [hunk 4]
```diff
@@ -263,6 +263,11 @@ def _(x):
 CELERY_RESULT_BACKEND = 'redis://localhost:6379/2'
 #: Scheduled tasks configuration (aka cronjobs).
 CELERY_BEAT_SCHEDULE = {
+    'scheduler-timestamp': {
```
Comment (jma) [inline rero_ils/config.py:266]: A small comment is very welcome here.

File: rero_ils/modules/monitoring.py [hunk 5]
```diff
@@ -270,6 +273,43 @@ def missing_pids(doc_type):
...
+                                 'redis://localhost:6379')
+    redis = Redis.from_url(url)
+    info = redis.info()
+    return jsonify({'data': info})
+
+
+@api_blueprint.route('/timestamps')
+@check_authentication
+def timestamps():
+    """Get time stamps from current cache.
```
Comment (jma) [inline rero_ils/modules/monitoring.py:293]: Can you explain a bit more what are timestamps?

File: rero_ils/modules/utils.py [hunk 6]
```diff
@@ -419,3 +421,43 @@ def get_patron_from_arguments(**kwargs):
         or Patron.get_patron_by_barcode(kwargs.get('patron_barcode')) \
         or Patron.get_record_by_pid(kwargs.get('patron_pid')) \
         or Patron.get_record_by_pid(kwargs.get('loan').get('patron_pid'))
+
+
+def set_timestamp(name, **kwargs):
+    """Set timestamp in current cache.
```
Comment (jma) [inline rero_ils/modules/utils.py:427]: Can you give more explanation? What is the purpose of this?

**Review by jma — APPROVED** [review]

**Comment by jma** [comment]
It would be great to have a description of the global idea. What is the relation of the timestamp and the monitoring?

File: rero_ils/config.py (hunk 4 above)
Comment (rerowep) [reply rero_ils/config.py:266]: added:
```
        # Save a timestamp so we can externaly test the timestamp changed
        # every minute. If the timestamp is not changing the scheduller
        # is not working.
```

File: rero_ils/modules/utils.py (hunk 6 above)
Comment (rerowep) [reply rero_ils/modules/utils.py:427]: added:
```
    Allows to timestamp functionality and monitoring of the changed
    timestamps externaly via url requests.
```

File: rero_ils/modules/monitoring.py (hunk 5 above)
Comment (rerowep) [reply rero_ils/modules/monitoring.py:293]: added:
```
Makes the timestamps accessible via url requests.
```

**Pull request merged**